fix(shared): update slack webhook configuration

AMarcosCastelo opened the pull request:
# Summary

Updates the Slack webhook configuration in GitHub Actions workflows to properly access the webhook URL from repository secrets.

# Details

The changes include:

1. **Security Enhancement**:

   - Changed `env.SLACK_WEBHOOK_URL` to `secrets.SLACK_WEBHOOK_URL` in workflow files
   - Ensures proper access to the secure repository variable
   - Affects both backend-check and deploy workflows

2. **Files Modified**:
   - `.github/workflows/backend-check.yaml`
   - `.github/workflows/deploy.yaml`

# Related Links

N/A

# Declaration

- [x] I have tested these changes and they work as expected
- [x] I have ensured no services are broken
- [x] I have followed best practices
- [x] The code quality has improved or remained the same


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Updated workflow configurations to improve security by sourcing Slack webhook URLs from repository secrets instead of variables for deployment and backend checks. No changes to workflow logic or notifications.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cris-santos commented:
Task linked: [CARROT-808 Configurar IAM assumeRole para o Elrond se conectar ao Smaug](https://app.clickup.com/t/3005225/CARROT-808)

cris-santos commented on libs/shared/rule/result/src/rule-result.helpers.ts:
```diff
@@ -42,19 +45,38 @@ export const mapToRuleOutput = (
   resultStatus,
 });
 
+export const assignRequest = async ({
```
fiquei um pouco confuso com o nome, não seria `signRequest`?

cris-santos commented on libs/shared/rule/result/src/rule-result.helpers.ts:
```diff
@@ -42,19 +45,38 @@ export const mapToRuleOutput = (
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    hostname: '',
+    method: 'POST',
+    path: '',
+    protocol: 'https',
```
só pra confirmar, precisamos passar os campos `hostname` e `path` vazios? já não é o valor padrão?

gabrielsl96 replied:
Pois é, são campos obrigatórios.

Poderia fazer um split na url para tratar esses campos, mas seria só um trabalho desnecessário

cris-santos commented on libs/shared/rule/result/src/rule-result.helpers.ts:
```diff
@@ -42,19 +45,38 @@ export const mapToRuleOutput = (
+  ruleOutput,
+}: {
+  ruleOutput: RuleOutput;
+}) => {
```
pra garantir, acho que valeria a pena incluir a tipagem de retorno

AMarcosCastelo committed:
fix(shared): update slack webhook configuration

Update GitHub Actions workflows to properly access SLACK_WEBHOOK_URL.

Use secrets.SLACK_WEBHOOK_URL instead of env.SLACK_WEBHOOK_URL for secure variable access.

AMarcosCastelo committed:
Merge branch 'main' into fix/slack-webhook-secret-config

AMarcosCastelo committed:
fix: add SLACK_WEBHOOK_URL as a required secret
